restapi/auth.go

```diff
@@ -11,38 +14,99 @@ import (
 	"github.com/uwblueprint/shoe-project/config"
 	"github.com/uwblueprint/shoe-project/internal/database/models"
 	"github.com/uwblueprint/shoe-project/restapi/rest"
-	"golang.org/x/crypto/bcrypt"
-	"gorm.io/gorm"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
 )
 
+var oauthconfig *oauth2.Config
+
+func generateStateOauthCookie(w http.ResponseWriter) (string, error) {
+	duration, err := config.GetTokenExpiryDuration()
+	if err != nil {
+		return "", err
+	}
+	var expiration = time.Now().Add(duration)
+
+	b := make([]byte, 16)
+	rand.Read(b)
+	state := base64.URLEncoding.EncodeToString(b)
+	fmt.Println(state)
+	cookie := http.Cookie{Name: "oauthstate", Value: state, Expires: expiration, HttpOnly: true} // in prod, set Secure to true
+	http.SetCookie(w, &cookie)
+
+	return state, nil
+}
+
 func (api api) Login(w http.ResponseWriter, r *http.Request) render.Renderer {
-	var user models.User
-	err := json.NewDecoder(r.Body).Decode(&user)
+	oauthconfig = &oauth2.Config{
```

Comment:
We can't declare this outside the function (maybe we should add it as a field to the api struct)

Reply:
good call